src/audio/google_rtc_audio_processing.c

```diff
@@ -122,10 +124,23 @@ static int google_rtc_audio_processing_cmd_set_data(
 	struct sof_ipc_ctrl_data *cdata)
 {
 	struct google_rtc_audio_processing_comp_data *cd = comp_get_drvdata(dev);
+	int ret;
 
 	switch (cdata->cmd) {
 	case SOF_CTRL_CMD_BINARY:
-		return comp_data_blob_set_cmd(cd->tuning_handler, cdata);
+		ret = comp_data_blob_set_cmd(cd->tuning_handler, cdata);
+		if (ret)
+			return ret;
+		/* Accept the new blob immediately so that userspace can write
+		 * the control in quick succession without error.
+		 * This ensures the last successful control write from userspace
+		 * before prepare/copy is applied.
+		 */
+		if (comp_is_new_data_blob_available(cd->tuning_handler)) {
+			comp_get_data_blob(cd->tuning_handler, NULL, NULL);
```

Comment:
Are we certain that we are not holding an references to the previous data at this point? The old data is freed here so if we did not dup that will be dangerous

Reply:
Added comment to explain why it is safe to free the old blob.